sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1660,17 +1614,14 @@ case class GpuSubstringIndex(strExpr: Expression,
   // spark-rapids plugin issue https://github.com/NVIDIA/spark-rapids/issues/8750
   override def doColumnar(str: GpuColumnVector, delim: GpuScalar,
       count: GpuScalar): ColumnVector = {
-    if (regexp == null) {
-      withResource(str.getBase.isNull) { isNull =>
-        withResource(Scalar.fromString("")) { emptyString =>
-          isNull.ifElse(str.getBase, emptyString)
-        }
-      }
-    } else {
-      withResource(str.getBase.extractRe(new RegexProgram(regexp))) { table: Table =>
-        table.getColumn(0).incRefCount()
-      }
+
+    val strColumnView = str.getBase
+    val delimiter = delim.getBase
+    val cnt = count.getValue.asInstanceOf[Int]
+    withResource(GpuSubstringIndexUtils.substringIndex(strColumnView, delimiter, cnt)) { result =>
```

Comment:
`result` is not an intermediate value so I think we don't need to manage it's resource. Just `GpuSubstringIndexUtils.substringIndex(str.getBase, delim.getBase, cnt)` will be work.